CorsixTH/Lua/hospitals/player_hospital.lua

```diff
@@ -815,6 +817,23 @@ function PlayerHospital:tickEarthquake(stage)
   end
 end
 
+--! Give advice that a patient is waiting for the player to build a GP's office
+-- Called when a patient has passed reception and is waiting for a room to be built.
+-- Each piece of advice is only given once per month
+function PlayerHospital:adviseNoGpsOffice()
+  if self:countStaffOfCategory("Doctor", 1) > 0 then -- Doctor without a room
+    if not self.adviser_data.gps_office_1 then
+      self.world.ui.adviser:say(_A.warnings.no_gps_office_2)
+      self.adviser_data.gps_office_1 = true
+    end
+  else -- No room or doctor
+    if not self.adviser_data.gps_office then
+      self.world.ui.adviser:say(_A.warnings.no_gps_office)
+      self.adviser_data.gps_office = true
```

Comment:
Any particular reason why the flag name must be different from the warning name?

Write boring code! Make the details really non-interesting details and very very very guessable.  
If you make both names the same, everybody understand how they are linked.

If you make them mostly the same but not exactly, and there are more of these, then soon it's a guessing game that takes much time to figure out, with all the risks of messing up.